### ISSUE TITLE
Different errors raised for similar cases

### ISSUE THREAD

**Issue by superbock** [olemb/mido#116]
`MidiFile.length` raises a `ValueError` whereas `MidiFile.__iter__` a `TypeError` in case of type 2 MIDI files. I think these should be unified, but I am not sure which one is more appropriate?

**Comment by cjappl** [comment]
Are Type 2 MIDI files not supported? 

If so maybe a custom error? "InvalidMidiFileType" or something? I find those are usually more specific than any builtin for something like this.

**Comment by superbock** [comment]
No, type 2 MIDI files are supported. Only iterating over all messages is not, because the individual tracks of these files are asynchronous and can thus not be combined.

As a side note, maybe adding a `__iter__` method to `MidiTrack` might be a good idea to iterate over the messages of a specific track.
